packages/screens/Organizations/components/RolesOrg/RolesSettingsSection.tsx

```diff
@@ -26,25 +29,119 @@ interface RolesSettingsSectionProps {
       setRolesIndexes(copyIndex);
     }
   };
 
+  const resetModal = () => {
+    resetField(`roles.${rolesIndexes.length}.name`);
+    resetField(`roles.${rolesIndexes.length}.color`);
+    resetField(`roles.${rolesIndexes.length}.resources`);
+  };
+
+  const onOpenModal = () => {
+    resetModal();
+    setResources(fakeResources.map((r) => ({ ...r, value: false })));
+    setModalVisible(true);
+  };
+
   const addRoleField = () => {
+    register(`roles.${rolesIndexes.length}.resources`);
+    const selectedResources = resources
+      .filter((r) => r.value)
+      .flatMap((r) => r.resources);
+    setValue(`roles.${rolesIndexes.length}.resources`, selectedResources);
+    console.log(`Selected resources: ${selectedResources}`);
     setRolesIndexes([...rolesIndexes, Math.floor(Math.random() * 200000)]);
+    setModalVisible(false);
+  };
+
+  const onCloseModal = () => {
+    setModalVisible(false);
+  };
+
+  const onCheckboxChange = (index: number) => {
+    const copyResources = [...resources];
+    copyResources[index].value = !copyResources[index].value;
+    setResources(copyResources);
   };
 
   return (
     <View style={{ flex: 1 }}>
+      <ModalBase
+        key={modalVisible ? "open" : "closed"}
+        visible={modalVisible}
+        onClose={onCloseModal}
+        width={480}
+        label="Add a new Role"
+      >
+        <View>
+          <TextInputCustom<RolesSettingFormType>
+            control={control}
+            noBrokenCorners
+            name={`roles.${rolesIndexes.length}.name`}
+            label="Role name"
+            placeholder="Role name"
+            rules={{ required: true }}
+            placeHolder="Role name"
+          />
+          <SpacerColumn size={2.5} />
+          <TextInputCustom<RolesSettingFormType>
+            control={control}
+            noBrokenCorners
+            name={`roles.${rolesIndexes.length}.color`}
+            label="Role color"
+            placeholder="Role color"
+            placeHolder="Role color"
+          />
+        </View>
+        <SpacerColumn size={2.5} />
+        <View>
+          <Label>Features & Permissions</Label>
+          <Box
+            style={{
+              borderWidth: 1,
+              borderColor: neutral33,
+              borderRadius: 8,
+              padding: 10,
+            }}
+          >
+            <ScrollView style={{ maxHeight: 180 }}>
+              {/* TODO: Refactor Checkbox to make it a global component instead of Dapp!*/}
+              {resources.map((resource, index) => (
+                <View style={{ flex: 1, flexDirection: "row" }}>
+                  <TouchableOpacity onPress={() => onCheckboxChange(index)}>
+                    <CheckboxDappStore isChecked={resource.value} />
+                  </TouchableOpacity>
+                  <SpacerRow size={1} />
+                  <BrandText style={fontSemibold18}>{resource.name}</BrandText>
+                  <SpacerColumn size={4} />
+                </View>
```

Comment:
You can `import Checkbox from "@/screens/Mini/components/Checkbox/Checkbox";
`
(Yes, we must export that from `Mini`, it makes non sense)

```suggestion
                <>
                <Checkbox
                  isChecked={resource.value}
                  value={resource.name}
                  label={resource.name}
                  onPress={() => onCheckboxChange(index)}
                />
                  <SpacerColumn size={1} />
                </>
```

Reply:
I think i tried the mini but it was smaller and looked weird but i will see

Reply:
[e254883](https://github.com/TERITORI/teritori-dapp/pull/1443/commits/e254883da1271e065957001bdce49470c48d5999)